src/components/tl/ucp/allgather/allgather_knomial.c

```diff
@@ -50,6 +51,9 @@ void ucc_tl_ucp_allgather_knomial_progress(ucc_coll_task_t *coll_task)
                                        args->root : 0;
     ucc_rank_t             rank      = VRANK(task->subset.myrank, broot, size);
     size_t                 local     = GET_LOCAL_COUNT(args, size, rank);
+    ucp_mem_h             *mh_list   = task->mh_list;
+    int                    max_count = task->count_mh;
+    int                    count_mh     = 0;
```

Comment:
BUG - this is a mistake, because u want to keep it when u go in and out from progress

Reply:
right but so how do I initialize it ?